app/download/views.py

```diff
@@ -114,6 +114,17 @@ def get_document_metadata(service_id, document_id):
 @download_blueprint.route("/services/<uuid:service_id>/documents/<uuid:document_id>/authenticate", methods=["POST"])
 def authenticate_access_to_document(service_id, document_id):
     key = request.json.get("key")
+    if redis_store.exceeded_rate_limit(
+        f"authenticate-document-{service_id}-{document_id}",
+        limit=current_app.config["DOCUMENT_AUTHENTICATION_RATE_LIMIT"],
+        interval=current_app.config["DOCUMENT_AUTHENTICATE_RATE_INTERVAL"],
+    ):
+        return (
```

Comment:
Optional, I'm not sure if this is a good idea or not: Do we also want to put in an info level log line here similar to the API? https://github.com/alphagov/notifications-api/blob/a81f34f8bd6a3fa44d358563464ae545e53ce758/app/notifications/validators.py#L59

Or maybe it is OK because unlike the API we know the url of the file from the logs where on the API we don't have anything to tell which service sent the request?

Reply:
In this case I don't think we need a specific log message, the URL gives us everything we need and it's explicit (ie not base64 encoded like the doc-download-frontend URLs are).